driver/src/test/java/org/neo4j/driver/v1/integration/SessionAsyncIT.java

```diff
@@ -1028,6 +1029,65 @@ public void shouldPropagateFailureFromFirstIllegalQuery()
         assertEquals( 0, countNodesByLabel( "Node3" ) );
     }
 
+    @Test
```

Comment:
I think we shall also add some more task chaining tests that mixes up sync / async usage with successful completion i.e. making use of `thenApplyAsync` to make chained task to be run in another executor? It may be good to  address this kind of usage in the docs.